#Baseline #4

#Number of orders: 1
#Number of shipments: 2
#Challenges:
#  -Interruption: No
#  -Drop parts: No
#  -Sensor blackout: No
#  -Part flipping: Yes
#  -Faulty parts: No
#  -Moving obstacle: No
#2 arms advantageous: No
#Containers
# -Belt: Yes
# -Bins: Optional
# -Shelf: Yes (shelf11 & shelf8)
#AGV
# -order_0:
#   -shipment_0: any
#   -shipment_1: any
#Time limit: No

time_limit: 500

options:
  gazebo_state_logging: false
  insert_models_over_bins: true
  insert_models_over_shelves: true
  belt_population_cycles: 15
  model_type_aliases:
    order_part1: pulley_part_blue
    order_part2: gear_part_green
    order_part3: disk_part_red
    order_part4: piston_rod_part_red

shelf_layout:
  row_1: ['base','collar',0,'base'] #shelves: 3, 4, 5
  row_2: ['pipe', 0,'base','base']   #shelves: 6, 7, 8
  row_3: ['base','pipe',0,'collar'] #shelves: 9, 10, 11


orders:
  order_0:
    announcement_condition: time
    announcement_condition_value: 0.0
    shipment_count: 2
    destinations: [any,any]
    products:
      part_0:
        type: pulley_part_blue
        pose:
          xyz: [0.0, 0.0, 0]
          rpy: ['pi', 0, 0]
      part_1:
        type: gear_part_green
        pose:
          xyz: [0.15, 0.2, 0]
          rpy: [0, 0, 'pi/4']
      part_2:
        type: disk_part_red
        pose:
          xyz: [0.15, -0.2, 0]
          rpy: [0, 0, '-pi/4']
      part_3:
        type: piston_rod_part_red
        pose:
          xyz: [-0.15, -0.2, 0]
          rpy: [0, 0, '-pi/4']

models_over_bins:
  bin13:
    models:
      disk_part_blue:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 3
  bin16:
    models:
      disk_part_green:
        xyz_start: [0.1, 0.1, 0.0]
        xyz_end: [0.5, 0.5, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 3
        num_models_y: 3
  # bin5:
  #   models:
  #     pulley_part_red:
  #       xyz_start: [0.15, 0.15, 0.0]
  #       xyz_end: [0.45, 0.45, 0.0]
  #       rpy: [0, 0, 0]
  #       num_models_x: 2
  #       num_models_y: 2
  bin8:
    models:
      pulley_part_red:
        xyz_start: [0.15, 0.15, 0.0]
        xyz_end: [0.45, 0.45, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 2
        num_models_y: 1
#2 arms advantageous
models_over_shelves:
  shelf5:
    models:
      piston_rod_part_red:
        xyz_start: [1, -0.4, 0.0]
        xyz_end: [0., -0.4, 0.0]
        rpy: [0, 0, 'pi/4']
        num_models_x: 6
        num_models_y: 1
  shelf8:
    models:
      disk_part_red:
        xyz_start: [1, -0.5, 0.0]
        xyz_end: [0., -0.4, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 5
        num_models_y: 2
  shelf11:
    models:
      gear_part_green:
        xyz_start: [1, 0.5, 0.0]
        xyz_end: [0., 0.4, 0.0]
        rpy: [0, 0, 0]
        num_models_x: 6
        num_models_y: 1

belt_models:
  pulley_part_blue:
    15.0:
      pose:
        xyz: [0.0, 4.3, 0.92]
        rpy: [0, 0, 0]


#Faulty products
faulty_products:
  - gear_part_green_3
  - disk_part_red_2
  - piston_rod_part_red_4
